TeamSnapSDK/SDK/CollectionJSON/TSDKCollectionObject.m

```diff
@@ -710,7 +715,10 @@ - (void)encodeWithCoder:(NSCoder *)coder {
 }
 
 - (BOOL)isNewObject {
-    return ([[self objectIdentifier] isEqualToString:@""] || [[self objectIdentifier] integerValue] == 0);
+    if (([[[self collection] data] objectForKey:@"id"] == nil) || [[[[[self collection] data] objectForKey:@"id"] description] isEqualToString:@""] || [[[[self collection] data] objectForKey:@"id"] isEqual:[NSNull null]] || ([[[[self collection] data] objectForKey:@"id"] integerValue] == 0)) {
```

Comment:
This could be simplified by caching `self.collection.data[@"id"]` and use it for comparison:

``` objc
id rawIdentifier = self.collection.data[@"id"]

if (rawIdentifier == nil ||
    [[rawIdentifier description] isEqualToString:@""] ||
    [rawIdentifier isEqual:[NSNull null]] ||
    [rawIdentifier integerValue] == 0) {
    return YES;
}
```